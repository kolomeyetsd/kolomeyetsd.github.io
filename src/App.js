//
import React, { Component } from 'react'
import { HashRouter, NavLink, Route } from 'react-router-dom'

import data from './data.json'

const BASE_FONT_SIZE = 10
const ADD_FONT_SIZE = 25


const App = props => {
  const { tags, maxSentiment } = props

  return (
    <HashRouter hashType="noslash">
      <div className="app">
        <div className="tags-side">
          {tags.map(tag => (
            <Tag
              key={tag.id}
              model={tag}
              maxSentiment={maxSentiment}
            />
          ))}
        </div>
        <div className="description-side">
          <Route
            path="/"
            exact={true}
            render={props => (
              <span>no selected tag</span>
            )}
          />
          <Route
            path="/:id+"
            render={props => (
              <Description model={tags.find(tag => tag.id === props.match.params.id)} />
            )}
          />
        </div>
      </div>
    </HashRouter>
  )
}


const Tag = props => {
  const { model, maxSentiment } = props
  const fontSize = BASE_FONT_SIZE + Math.round(model.sentimentScore / maxSentiment * ADD_FONT_SIZE)

  return (
    <NavLink activeClassName="tag-active" className="tag" to={`/${model.id}`} style={{ fontSize }}>
      {props.model.label}
    </NavLink>
  )
}


const Description = props => {
  const { model } = props

  const totalMentions = Object
    .entries(model.sentiment)
    .reduce((total, [ key, val ]) => total + val, 0)

  const pageTypes = Object
    .entries(model.pageType)
    .filter(([ key, val ]) => val > 0)
    .reduce((out, [ key, val ]) => [ ...out, { name: key, value: val } ], [])
    .sort((a, b) => b.value - a.value)

  return (
    <div className="description">
      <span className="label">{model.label}</span>
      <ul>
        <li>
          {`total mentions: ${totalMentions}`}
        </li>
        <li>
          {`positive mentions: ${model.sentiment.positive || 'none'}`}
        </li>
        <li>
          {`neutral mentions: ${model.sentiment.neutral || 'none'}`}
        </li>
        <li>
          {`negative mentions: ${model.sentiment.negative || 'none'}`}
        </li>
        <li>
          <span>page types</span>
          <ul>
            {pageTypes.map(type => (
              <li key={type.name}>
                {`${type.name}: ${type.value}`}
              </li>
            ))}
          </ul>
        </li>
      </ul>
    </div>
  )
}


export default props => (
  <App
    tags={data}
    maxSentiment={Math.max(...data.map(tag => tag.sentimentScore))}
  />
)
