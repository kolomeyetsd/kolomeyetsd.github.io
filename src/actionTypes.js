//
export const STATE_RESET = 'STATE_RESET'
export const STATE_LOADED = 'STATE_LOADED'
export const PIECE_DRAG_STARTED = 'PIECE_DRAG_STARTED'
export const SQUARE_DRAG_OVER = 'SQUARE_DRAG_OVER'
export const PIECE_DROPPED = 'PIECE_DROPPED'
