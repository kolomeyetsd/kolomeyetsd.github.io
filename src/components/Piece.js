//
import React, { Component } from 'react'
import { connect } from 'react-redux'
import classnames from 'classnames'

const SPRITE_PATH = '/public/chess-sprite.svg'


export class Piece extends Component {

  constructor (props) {
    super(props)

    this.state = {}

    this.onMouseDown = this.onMouseDown.bind(this)
    this.onMouseMove = this.onMouseMove.bind(this)
    this.onMouseUp = this.onMouseUp.bind(this)
  }

  onMouseDown (e) {
    if (this.props.model.color !== this.props.turnColor) return

    e.preventDefault()
    e.stopPropagation()

    document.addEventListener('mousemove', this.onMouseMove)
    document.addEventListener('mouseup', this.onMouseUp)

    const startX = e.pageX
    const startY = e.pageY

    this.setState({ startX, startY })
    this.props.onDragStart()
  }

  onMouseMove (e) {
    e.stopPropagation()
    e.preventDefault()

    this.setState({
      dragging: true,
      left: e.pageX - this.state.startX,
      top: e.pageY - this.state.startY
    })
  }

  onMouseUp (e) {
    e.stopPropagation()
    e.preventDefault()

    document.removeEventListener('mousemove', this.onMouseMove)
    document.removeEventListener('mouseup', this.onMouseUp)

    this.setState({ dragging: false })
    this.props.onDrop()
  }

  render () {
    const { props, state } = this
    const { model } = props

    const className = classnames(
      'piece',
      {'no-user-drag': this.props.model.color !== this.props.turnColor}
    )
    const style = !state.dragging ? {} : {
      pointerEvents: 'none',
      transform: `translate(${state.left}px, ${state.top}px)`,
      zIndex: 1
    }

    return (
      <svg className={className} style={style} onMouseDown={this.onMouseDown}>
        <use xlinkHref={`${SPRITE_PATH}#${model.color}-${model.name}`} />
      </svg>
    )
  }
}

export const PieceContainer = connect(
  (state, ownProps) => {
    return {
      ...ownProps,
      turnColor: state.turnColor
    }
  }
)(Piece)
