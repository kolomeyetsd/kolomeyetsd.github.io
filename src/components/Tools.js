//
import React, { Component } from 'react'
import { connect } from 'react-redux'

import * as actionTypes from '../actionTypes'

const LOCAL_STORAGE_KEY = 'chess_state'


class Tools extends Component {

  constructor (props) {
    super(props)

    this.state = {
      canLoad: localStorage.getItem(LOCAL_STORAGE_KEY)
    }

    this.onButtonClick = this.onButtonClick.bind(this)
  }

  onButtonClick (action) {
    switch (action) {

      case 'save': {
        localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(this.props, null, 2))
        this.setState({ canLoad: true })
      }
      break

      case 'load': {
        this.props.dispatch({
          type: actionTypes.STATE_LOADED,
          state: JSON.parse(localStorage.getItem(LOCAL_STORAGE_KEY))
        })
      }
      break

      case 'reset': {
        this.props.dispatch({ type: actionTypes.STATE_RESET })
      }
      break
    }
  }

  render () {
    return (
      <div className="app-tools">
        <span>{`Turn color: ${this.props.turnColor}`}</span>
        <Button
          onClick={this.onButtonClick}
          action="save"
          disabled={this.props.turnsCount === 0}
        />
        <Button
          onClick={this.onButtonClick}
          action="load"
          disabled={!this.state.canLoad}
        />
        <Button
          onClick={this.onButtonClick}
          action="reset"
          disabled={this.props.turnsCount === 0}
        />
      </div>
    )
  }
}

const Button = props => {
  if (props.disabled) {
    return (
      <span>
        {props.action}
      </span>
    )
  }

  return (
    <a href="#" onClick={() => props.onClick(props.action)}>
      {props.action}
    </a>
  )
}


export const ToolsContainer = connect(state => state)(Tools)
