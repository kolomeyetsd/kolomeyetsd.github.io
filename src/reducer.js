//
import * as actionTypes from './actionTypes'
import {
  isTurnLegal,
  isSquareBusy,
  initGameState,
  switchTurnColor
} from './gameLogicHelpers'



export default (state = initGameState(), action) => {

  switch (action.type) {

    case actionTypes.STATE_RESET: {
      return initGameState()
    }

    case actionTypes.STATE_LOADED: {
      return action.state
    }

    case actionTypes.PIECE_DRAG_STARTED: {
      return {
        ...state,
        dragFromSquareId: action.square.id,
        dragOverSquareId: action.square.id,
        allowedDropSquareIds: state.squares
          .filter(square => isTurnLegal(action.square, square, state.turnColor))
          .filter(square => !isSquareBusy(square, state.turnColor))
          .map(square => square.id),
      }
    }

    case actionTypes.SQUARE_DRAG_OVER: {
      return {
        ...state,
        dragOverSquareId: action.square.id
      }
    }

    case actionTypes.PIECE_DROPPED: {
      const { dragOverSquareId, allowedDropSquareIds } = state

      if ((action.square.id === dragOverSquareId) || !allowedDropSquareIds.includes(dragOverSquareId)) {
        return {
          ...state,
          allowedDropSquareIds: null
        }
      }

      const turnColor = switchTurnColor(state.turnColor)
      const { piece } = state.squares.find(square => square.id === state.dragFromSquareId)

      return {
        turnsCount: state.turnsCount + 1,
        turnColor,
        allowedDropSquareIds: null,
        dragOverSquareId: null,
        dragOverSquareId: null,
        squares: state.squares.map(square => {
          if (square.id === state.dragOverSquareId) {
            return {
              ...square,
              piece: {
                ...piece,
                turnsCount: piece.turnsCount + 1
              }
            }
          }
          if (square.id === state.dragFromSquareId) return { ...square, piece: null }
          return square
        })
      }
    }

    default: return state
  }
}
