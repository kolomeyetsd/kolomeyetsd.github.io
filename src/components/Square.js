//
import React, { Component } from 'react'
import { connect } from 'react-redux'
import classnames from 'classnames'

import { PieceContainer } from './Piece'
import * as actionTypes from '../actionTypes'


export class Square extends Component {

  constructor (props) {
    super(props)

    this.onMouseOver = this.onMouseOver.bind(this)
    this.onPieceDragStart = this.onPieceDragStart.bind(this)
    this.onPieceDrop = this.onPieceDrop.bind(this)
  }

  onMouseOver (e) {
    if (!this.props.allowedDropSquareIds) return

    this.props.dispatch({
      type: actionTypes.SQUARE_DRAG_OVER,
      square: this.props.model
    })
  }

  onPieceDragStart () {
    this.props.dispatch({
      type: actionTypes.PIECE_DRAG_STARTED,
      square: this.props.model
    })
  }

  onPieceDrop () {
    this.props.dispatch({
      type: actionTypes.PIECE_DROPPED,
      square: this.props.model
    })
  }

  render () {
    const { props, state } = this
    const { model } = props

    const highlightColor = getHighlightColor(
      props.allowedDropSquareIds,
      props.dragOverSquareId,
      props.dragFromSquareId,
      model.id
    )
    const className = classnames('square', model.black ? 'black' : 'white')

    return (
      <div className={className} onMouseOver={this.onMouseOver}>
        {highlightColor && (
          <div className={classnames('highlight', highlightColor)} />
        )}
        {model.piece && (
          <PieceContainer
            model={model.piece}
            onDragStart={this.onPieceDragStart}
            onDrag={this.onPieceDrag}
            onDrop={this.onPieceDrop}
          />
        )}
      </div>
    )
  }
}


export const SquareContainer = connect(
  (state, ownProps) => {
    return {
      ...ownProps,
      dragOverSquareId: state.dragOverSquareId,
      dragFromSquareId: state.dragFromSquareId,
      allowedDropSquareIds: state.allowedDropSquareIds
    }
  }
)(Square)


const getHighlightColor = (allowedIds, dragOverId, dragFromId, targetId) => {
  if (!allowedIds) return false
  if (allowedIds.includes(targetId)) return dragOverId === targetId ? 'green' : 'blue'
  if (dragOverId === targetId) return dragFromId === targetId ? 'grey' : 'red'
  return false
}
