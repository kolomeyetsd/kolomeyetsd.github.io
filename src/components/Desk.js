//
import React from 'react'
import { connect } from 'react-redux'

import { SquareContainer } from './Square'


const Desk = props => {
  return (
    <div className="desk">
      {props.squares.map((square, i) => (
        <SquareContainer key={i} index={i} model={square} />
      ))}
    </div>
  )
}

export const DeskContainer = connect(
  (state) => {
    return {
      squares: state.squares
    }
  }
)(Desk)
