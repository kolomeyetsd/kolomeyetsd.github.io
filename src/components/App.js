//
import React, { Component, Fragment } from 'react'
import { Provider, connect } from 'react-redux'
import { createStore } from 'redux'

import { DeskContainer } from './Desk'
import { ToolsContainer } from './Tools'
import reducer from '../reducer'

const store = createStore(reducer)

export const App = props => {
  return (
    <Provider store={store}>
      <DeskContainer />
      <ToolsContainer />
    </Provider>
  )
}
