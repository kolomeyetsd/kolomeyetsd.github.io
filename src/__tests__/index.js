//
import React from 'react'
import renderer from 'react-test-renderer'

import { App } from '../components/App'
import reducer from '../reducer'
import * as actionTypes from '../actionTypes'


describe('test just one render case', () => {

  it('should render app tree', () => {
    const tree = renderer.create(<App />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})


describe('test some reducer cases', () => {
  const state = reducer(undefined, { type: actionTypes.STATE_RESET })
  const squareId = 'squareId'
  const square = {
    id: squareId,
    x: 1,
    y: 7,
    piece: { id: 'pieceId', name: 'knight', color: 'black' }
  }
  const pieceDragStartedAction = { type: actionTypes.PIECE_DRAG_STARTED, square }
  const pieceDropAction = { type: actionTypes.PIECE_DROPPED, square }

  it('should set proper square ids: fromId, toId, allowedToDropIds - when black knight drag started', () => {
    const nextState = reducer(state, pieceDragStartedAction)

    expect(nextState.dragFromSquareId).toBe(squareId)
    expect(nextState.dragFromSquareId).toBe(squareId)
    expect(nextState.allowedDropSquareIds).toEqual([ 40, 42 ])
  })

  it('should not consider as turn, when drop piece on square from which turn started', () => {
    const nextState = reducer(reducer(state, pieceDragStartedAction), pieceDropAction)

    expect(nextState.turnsCount).toBe(0)
  })
})
