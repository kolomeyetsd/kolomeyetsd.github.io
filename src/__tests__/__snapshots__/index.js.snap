// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test just one render case should render app tree 1`] = `
Array [
  <div
    className="desk"
  >
    <div
      className="square white"
      onMouseOver={[Function]}
    >
      <svg
        className="piece no-user-drag"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#white-rook"
        />
      </svg>
    </div>
    <div
      className="square black"
      onMouseOver={[Function]}
    >
      <svg
        className="piece no-user-drag"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#white-knight"
        />
      </svg>
    </div>
    <div
      className="square white"
      onMouseOver={[Function]}
    >
      <svg
        className="piece no-user-drag"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#white-bishop"
        />
      </svg>
    </div>
    <div
      className="square black"
      onMouseOver={[Function]}
    >
      <svg
        className="piece no-user-drag"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#white-king"
        />
      </svg>
    </div>
    <div
      className="square white"
      onMouseOver={[Function]}
    >
      <svg
        className="piece no-user-drag"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#white-queen"
        />
      </svg>
    </div>
    <div
      className="square black"
      onMouseOver={[Function]}
    >
      <svg
        className="piece no-user-drag"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#white-bishop"
        />
      </svg>
    </div>
    <div
      className="square white"
      onMouseOver={[Function]}
    >
      <svg
        className="piece no-user-drag"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#white-knight"
        />
      </svg>
    </div>
    <div
      className="square black"
      onMouseOver={[Function]}
    >
      <svg
        className="piece no-user-drag"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#white-rook"
        />
      </svg>
    </div>
    <div
      className="square black"
      onMouseOver={[Function]}
    >
      <svg
        className="piece no-user-drag"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#white-pawn"
        />
      </svg>
    </div>
    <div
      className="square white"
      onMouseOver={[Function]}
    >
      <svg
        className="piece no-user-drag"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#white-pawn"
        />
      </svg>
    </div>
    <div
      className="square black"
      onMouseOver={[Function]}
    >
      <svg
        className="piece no-user-drag"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#white-pawn"
        />
      </svg>
    </div>
    <div
      className="square white"
      onMouseOver={[Function]}
    >
      <svg
        className="piece no-user-drag"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#white-pawn"
        />
      </svg>
    </div>
    <div
      className="square black"
      onMouseOver={[Function]}
    >
      <svg
        className="piece no-user-drag"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#white-pawn"
        />
      </svg>
    </div>
    <div
      className="square white"
      onMouseOver={[Function]}
    >
      <svg
        className="piece no-user-drag"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#white-pawn"
        />
      </svg>
    </div>
    <div
      className="square black"
      onMouseOver={[Function]}
    >
      <svg
        className="piece no-user-drag"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#white-pawn"
        />
      </svg>
    </div>
    <div
      className="square white"
      onMouseOver={[Function]}
    >
      <svg
        className="piece no-user-drag"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#white-pawn"
        />
      </svg>
    </div>
    <div
      className="square white"
      onMouseOver={[Function]}
    />
    <div
      className="square black"
      onMouseOver={[Function]}
    />
    <div
      className="square white"
      onMouseOver={[Function]}
    />
    <div
      className="square black"
      onMouseOver={[Function]}
    />
    <div
      className="square white"
      onMouseOver={[Function]}
    />
    <div
      className="square black"
      onMouseOver={[Function]}
    />
    <div
      className="square white"
      onMouseOver={[Function]}
    />
    <div
      className="square black"
      onMouseOver={[Function]}
    />
    <div
      className="square black"
      onMouseOver={[Function]}
    />
    <div
      className="square white"
      onMouseOver={[Function]}
    />
    <div
      className="square black"
      onMouseOver={[Function]}
    />
    <div
      className="square white"
      onMouseOver={[Function]}
    />
    <div
      className="square black"
      onMouseOver={[Function]}
    />
    <div
      className="square white"
      onMouseOver={[Function]}
    />
    <div
      className="square black"
      onMouseOver={[Function]}
    />
    <div
      className="square white"
      onMouseOver={[Function]}
    />
    <div
      className="square white"
      onMouseOver={[Function]}
    />
    <div
      className="square black"
      onMouseOver={[Function]}
    />
    <div
      className="square white"
      onMouseOver={[Function]}
    />
    <div
      className="square black"
      onMouseOver={[Function]}
    />
    <div
      className="square white"
      onMouseOver={[Function]}
    />
    <div
      className="square black"
      onMouseOver={[Function]}
    />
    <div
      className="square white"
      onMouseOver={[Function]}
    />
    <div
      className="square black"
      onMouseOver={[Function]}
    />
    <div
      className="square black"
      onMouseOver={[Function]}
    />
    <div
      className="square white"
      onMouseOver={[Function]}
    />
    <div
      className="square black"
      onMouseOver={[Function]}
    />
    <div
      className="square white"
      onMouseOver={[Function]}
    />
    <div
      className="square black"
      onMouseOver={[Function]}
    />
    <div
      className="square white"
      onMouseOver={[Function]}
    />
    <div
      className="square black"
      onMouseOver={[Function]}
    />
    <div
      className="square white"
      onMouseOver={[Function]}
    />
    <div
      className="square white"
      onMouseOver={[Function]}
    >
      <svg
        className="piece"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#black-pawn"
        />
      </svg>
    </div>
    <div
      className="square black"
      onMouseOver={[Function]}
    >
      <svg
        className="piece"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#black-pawn"
        />
      </svg>
    </div>
    <div
      className="square white"
      onMouseOver={[Function]}
    >
      <svg
        className="piece"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#black-pawn"
        />
      </svg>
    </div>
    <div
      className="square black"
      onMouseOver={[Function]}
    >
      <svg
        className="piece"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#black-pawn"
        />
      </svg>
    </div>
    <div
      className="square white"
      onMouseOver={[Function]}
    >
      <svg
        className="piece"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#black-pawn"
        />
      </svg>
    </div>
    <div
      className="square black"
      onMouseOver={[Function]}
    >
      <svg
        className="piece"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#black-pawn"
        />
      </svg>
    </div>
    <div
      className="square white"
      onMouseOver={[Function]}
    >
      <svg
        className="piece"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#black-pawn"
        />
      </svg>
    </div>
    <div
      className="square black"
      onMouseOver={[Function]}
    >
      <svg
        className="piece"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#black-pawn"
        />
      </svg>
    </div>
    <div
      className="square black"
      onMouseOver={[Function]}
    >
      <svg
        className="piece"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#black-rook"
        />
      </svg>
    </div>
    <div
      className="square white"
      onMouseOver={[Function]}
    >
      <svg
        className="piece"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#black-knight"
        />
      </svg>
    </div>
    <div
      className="square black"
      onMouseOver={[Function]}
    >
      <svg
        className="piece"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#black-bishop"
        />
      </svg>
    </div>
    <div
      className="square white"
      onMouseOver={[Function]}
    >
      <svg
        className="piece"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#black-king"
        />
      </svg>
    </div>
    <div
      className="square black"
      onMouseOver={[Function]}
    >
      <svg
        className="piece"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#black-queen"
        />
      </svg>
    </div>
    <div
      className="square white"
      onMouseOver={[Function]}
    >
      <svg
        className="piece"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#black-bishop"
        />
      </svg>
    </div>
    <div
      className="square black"
      onMouseOver={[Function]}
    >
      <svg
        className="piece"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#black-knight"
        />
      </svg>
    </div>
    <div
      className="square white"
      onMouseOver={[Function]}
    >
      <svg
        className="piece"
        onMouseDown={[Function]}
        style={Object {}}
      >
        <use
          xlinkHref="/public/chess-sprite.svg#black-rook"
        />
      </svg>
    </div>
  </div>,
  <div
    className="app-tools"
  >
    <span>
      Turn color: black
    </span>
    <span>
      save
    </span>
    <span>
      load
    </span>
    <span>
      reset
    </span>
  </div>,
]
`;
